presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralEncoder.java

```diff
@@ -257,9 +258,38 @@ else if (type instanceof RowType) {
             return rowType.getTypeParameters().stream()
                     .allMatch(LiteralEncoder::isSupportedLiteralType);
         }
+        else if (type instanceof MapType) {
+            MapType mapType = (MapType) type;
+            return isSupportedLiteralType(mapType.getKeyType()) && isSupportedLiteralType(mapType.getValueType());
+        }
         return SUPPORTED_LITERAL_TYPES.contains(type.getJavaType());
     }
 
+    public static long estimatedSizeInBytes(Object object)
+    {
+        if (object == null) {
+            return 1;
+        }
+        Class<?> javaType = object.getClass();
+        if (javaType == Long.class) {
+            return Long.BYTES;
+        }
+        else if (javaType == Double.class) {
+            return Double.BYTES;
+        }
+        else if (javaType == Boolean.class) {
+            return 1;
+        }
+        else if (object instanceof Block) {
+            return ((Block) object).getSizeInBytes();
+        }
+        else if (object instanceof Slice) {
+            return ((Slice) object).getRetainedSize();
```

Comment:
nit: would `getLenght` a better estimation for serialized size?

Reply:
Thanks, checked again the definition and `getRetainedSize` might include the full size of referenced sequence (Array/String).  `getLength` actually returns size in bytes so is what we wanted.